server/protocols/webadmin/webadmin-rabbitmq/src/main/java/org/apache/james/webadmin/routes/RabbitMQRoutes.java

```diff
@@ -0,0 +1,165 @@
+
+    @Inject
+    @SuppressWarnings("unchecked")
+    @VisibleForTesting
+    RabbitMQRoutes(MailQueueFactory<? extends ManageableMailQueue> mailQueueFactory,
+                   Clock clock, JsonTransformer jsonTransformer, TaskManager taskManager) {
+        this.mailQueueFactory = mailQueueFactory;
+        this.clock = clock;
+        this.jsonTransformer = jsonTransformer;
+        this.taskManager = taskManager;
+    }
+
+    @Override
+    public String getBasePath() {
+        return BASE_URL;
+    }
+
+    @Override
+    public void define(Service service) {
+        rebuildMailQueue(service);
+    }
+
+
+    @POST
+    @Path("/spooler")
+    @ApiOperation(
+        value = "rebuild the RabbitMQ MailQueue using the cassandra mail queue view"
+    )
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.CREATED_201, message = "OK, the task for rebuilding the queue is created"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid request for rebuilding the mail queue."),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
+    })
+    public void rebuildMailQueue(Service service) {
+        TaskFromRequest taskFromRequest = this::rebuildMailQueue;
+        service.post(BASE_URL + "/spooler",
+            taskFromRequest.asRoute(taskManager),
+            jsonTransformer);
+    }
+
+    private Task rebuildMailQueue(Request request) {
+        checkTask(request, "rebuild");
+        return new RebuildMailQueueTask(getSpoolerMailQueue(), getOlderThan(request));
+    }
+
+    private RabbitMQMailQueue getSpoolerMailQueue() {
+        ManageableMailQueue fetchedMailQueue = mailQueueFactory.getQueue(MailQueueFactory.SPOOL)
+            .orElseThrow(() -> ErrorResponder.builder()
+                .message("Spooler MailQueue not created")
+                .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
+                .type(ErrorResponder.ErrorType.SERVER_ERROR)
+                .haltError());
+
+        if (!(fetchedMailQueue instanceof RabbitMQMailQueue)) {
+            throw ErrorResponder.builder()
+                .message("Spooler MailQueue is not a RabbitMQ queue")
+                .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
+                .type(ErrorResponder.ErrorType.SERVER_ERROR)
+                .haltError();
+        }
+
+        return (RabbitMQMailQueue) fetchedMailQueue;
+    }
+
+    private void checkTask(Request request, String expectedTaskName) {
+        Optional.ofNullable(request.queryParams("task"))
+            .filter(Predicate.isEqual(expectedTaskName))
+            .orElseThrow(() -> ErrorResponder.builder()
+                .message("Task unknown")
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .haltError());
+    }
+
+    private Instant getOlderThan(Request req) {
```

Comment:
getOlderThan is a terrible name of method.

Reply:
do you have a proposition?